src/app/api/file/route.ts

```diff
@@ -0,0 +1,362 @@
+        parents: [parentID],
+      };
+
+      const fileCreateData = {
+        resource: fileMetadata,
+        fields: "id",
+      };
+
+      // NOTE: File will still be created on Drive even if it fails on MongoDB
+      const file = await (service as NonNullable<typeof service>).files.create(
+        fileCreateData
+      );
+
+      const googleFileId = file.data.id;
+
+      // If the data is valid, save it to the database via prisma client
+      await prisma?.file.create({
+        data: {
+          date: fileData.date,
+          filetype: fileData.filetype,
+          url: `https://docs.google.com/document/d/${googleFileId}`,
+          seniorId: fileData.seniorId,
+          Tags: fileData.Tags,
+        },
+      });
+
+      return NextResponse.json(
+        FileResponse.parse({
+          code: "SUCCESS",
+          message: "File successfully added",
+        }),
+        { status: 200 }
+      );
+    }
+  } catch (e) {
+    console.log("Error:", e);
+    return NextResponse.json(
+      FileResponse.parse({
+        code: "UNKNOWN",
+        message: "Unknown error received",
+      }),
+      { status: 500 }
+    );
+  }
+});
+
+export const PATCH = withSession(async (request) => {
+  try {
+    const { service, fileRequest } = await initializeDriveAuth(request);
+
+    if (!fileRequest.success) {
+      console.log(fileRequest.error);
+      return NextResponse.json(
+        FileResponse.parse({
+          code: "NOT_FOUND",
+          message: "Unsuccessful request creation",
+        }),
+        { status: 400 }
+      );
+    } else {
+      const fileData = fileRequest.data;
+
+      // Check that user has this senior assigned to them
+      const { SeniorIDs } = await prisma.user.findFirst({
+        where: {
+          id: request.session.user.id,
+        },
+      });
+
+      if (
+        !SeniorIDs.some((seniorId: string) => seniorId === fileData.seniorId)
+      ) {
+        return NextResponse.json(
+          FileResponse.parse({
+            code: "NOT_AUTHORIZED",
+            message: "Senior not assigned to user",
+          }),
+          { status: 404 }
+        );
+      }
+
+      // Get senior from database
+      const foundSenior = await prisma.senior.findUnique({
+        where: { id: fileData.seniorId },
+      });
+      if (foundSenior == null) {
+        return NextResponse.json(
+          FileResponse.parse({
+            code: "NOT_FOUND",
+            message: "Senior not found",
+          }),
+          { status: 404 }
+        );
+      }
+
+      const formatted_date =
+        (fileData.date.getMonth() > 8
+          ? fileData.date.getMonth() + 1
+          : "0" + (fileData.date.getMonth() + 1)) +
+        "/" +
+        (fileData.date.getDate() > 9
+          ? fileData.date.getDate()
+          : "0" + fileData.date.getDate()) +
+        "/" +
+        fileData.date.getFullYear();
+
+      // Used for extracting out the fileId from the url
+      const pattern =
```

Comment:
Can we potentially simplify this logic by passing the target `fileId` when calling the request? Maybe we can explore `file/[fileId]` path or pass the field directly in the body.